### PR TITLE
feat(module:notification): add close icon prop

### LINKED ISSUE #4494: add closeIcon API to Notification component
Ref https://github.com/ant-design/ant-design/releases/tag/3.26.0.

### PR DESCRIPTION
**Pull request by wendellhu95**
close #4494

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #4494

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/notification/nz-notification.component.html [hunk 1]
```diff
@@ -25,7 +25,14 @@
   </ng-template>
   <a tabindex="0" class="ant-notification-notice-close" (click)="close()">
     <span class="ant-notification-notice-close-x">
-      <i nz-icon nzType="close" class="ant-notification-close-icon"></i>
+      <ng-container *ngIf="nzMessage.options?.nzCloseIcon">
+        <ng-container *nzStringTemplateOutlet="nzMessage.options?.nzCloseIcon">
+          <i nz-icon [nzType]="nzMessage.options?.nzCloseIcon"></i>
```
Comment (hsuanxyz) [inline components/notification/nz-notification.component.html]: 上面安全取值了，下面的 `?.` 操作符是不是可以不要了

Comment (wendzhue) [reply components/notification/nz-notification.component.html]: 不行，试过了

File: components/notification/nz-notification.component.html [hunk 2]
```diff
@@ -25,7 +25,14 @@
...
+        </ng-container>
+      </ng-container>
+      <ng-container *ngIf="!nzMessage.options?.nzCloseIcon">
```
Comment (vthinkxie) [inline components/notification/nz-notification.component.html]: 用 else?

Comment (wendzhue) [reply components/notification/nz-notification.component.html]: 也行……

**Pull request merged**